src/views/Leads/ActionButtons/LeadEditModal/index.tsx

```diff
@@ -0,0 +1,191 @@
+    patchLead: (leadPatchParams: LeadPatchParams) => void;
+}
+
+interface Params {
+    leadId?: number;
+    body?: LeadFaramValues;
+}
+
+const requestOptions: Requests<OwnProps, Params> = {
+    leadEditRequest: {
+        url: ({ params }) => `/v2/leads/${params && params.leadId}/`,
+        body: ({ params }) => params && params.body,
+        method: methods.PUT,
+        onSuccess: ({ response, props }) => {
+            const lead = response as LeadFaramValues;
+            const {
+                patchLead,
+            } = props;
+
+            patchLead({ lead });
+            notify.send({
+                type: notify.type.SUCCESS,
+                title: _ts('leads', 'leads'),
+                message: _ts('leads', 'leadEditedSuccessfullyMessage'),
+                duration: notify.duration.FAST,
+            });
+        },
+        onFailure: notifyOnFailure(_ts('leads', 'leads')),
+        onFatal: notifyOnFatal(_ts('leads', 'leads')),
+    },
+};
+
+// TODO: Write typings for dispatch
+const mapDispatchToProps = dispatch => ({
+    patchLead: params => dispatch(patchLeadAction(params)),
+});
+
+type Props = AddRequestProps<OwnProps, Params>;
+
+function LeadEditModal(props: Props) {
+    const {
+        leadId,
+        lead: leadFromProps,
+        closeModal,
+        requests: {
+            leadEditRequest: {
+                do: onLeadEdit,
+                pending,
+            },
+        },
+    } = props;
+    console.warn('here', leadFromProps);
+
+    const [leadFaramValues, setLeadFaramValues] = useState<LeadFaramValues>(leadFromProps);
+    const [leadFaramErrors, setLeadFaramErrors] = useState<LeadFaramErrors>({});
+    const [pristine, setPristine] = useState<LeadFaramErrors>({});
+
+    const handleLeadDetailChange = useCallback(({
+        faramValues: newFaramValues,
+        faramErrors: newFaramErrors,
+    }) => {
+        setLeadFaramValues(newFaramValues);
+        setLeadFaramErrors(newFaramErrors);
+        setPristine(false);
+    }, [setLeadFaramValues, setLeadFaramErrors, setPristine]);
+
+    const handleFormSubmit = useCallback(() => {
+        detachedFaram({
+            value: leadFaramValues,
+            schema,
+            onValidationFailure: (faramErrors: LeadFaramErrors) => {
+                setLeadFaramErrors(faramErrors);
+            },
+            onValidationSuccess: (faramValues: LeadFaramValues) => {
+                onLeadEdit({
+                    leadId,
+                    body: faramValues,
+                });
+            },
+        });
+    }, [
+        leadFaramValues,
+        setLeadFaramErrors,
+        onLeadEdit,
+        leadId,
+    ]);
+
+    const lead = useMemo(() => ({
+        faramValues: leadFaramValues,
+        faramErrors: leadFaramErrors,
+    }), [leadFaramValues, leadFaramErrors]);
+
+    const isSubmitDisabled = useMemo(() => (
+        !doesObjectHaveNoData(leadFaramErrors) || pristine
+    ), [pristine, leadFaramErrors]);
+
```

Comment:
There are certain cases when this should not be done.
I forgot the cases though